src/StatsdClient/Statsd.cs

```diff
@@ -329,10 +329,11 @@ public void Send(string command)
         {
             try
             {
-                Udp.Send(command);
                 // clear buffer (keep existing behavior)
                 if (Commands.Count > 0)
                     Commands = new List<string>();

```

Comment:
https://github.com/DataDog/dogstatsd-csharp-client/commit/4d03ca463e94785e3d5deff064b499092d2f865e: It was to keep the same behavior as before but I think it could be removed.

Reply:
Now, buffering the http requests requires to call `Add` several times and then `Send` which is not efficient and not user friendly. I am planning to do something similar to the Go client which does the buffering automatically. So this logic will be completely rewritten.

Reply:
There is an explicit unit test https://github.com/DataDog/dogstatsd-csharp-client/blob/master/tests/StatsdClient.Tests/StatsdUnitTests.cs#L487-L497 that check this behavior. Keep it for now.